the_wizz/pair_maker.py

```diff
@@ -123,19 +128,15 @@ def run(self, reference_catalog, unknown_catalog, use_unkn_weights=False):
             self.hdf5_writer = Pool(1)
 
         print("Starting iteration...")
-        for ref_vect, redshift, dist, ref_id in zip(ref_vects,
-                                                    redshifts,
-                                                    dists,
-                                                    ref_ids):
+        ref_obj_iter = zip(
+            reference_catalog["ra"], reference_catalog["dec"],
```

Comment:
The zip is working here, however, you haven't masked the ra/dec values on redshift so the locations of some objects aren't lining up for some of the redshifts and distances.